.github/workflows/my-pipeline.yml

```
@@ -0,0 +1,20 @@
+name: Continuous Integration
+
+on:
+  push:
+    paths-ignore:
+      - 'ansible'
+      - 'README.md'
+  pull_request:
+    paths-ignore:
+      - 'ansible'
+      - 'README.md'
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```

Minor (and I know we're out of date in some places!) but checkout is actually on v4 now which should fix up some of your warnings:
```suggestion
    - uses: actions/checkout@v4
```